src/System.Reflection.MetadataLoadContext/src/System/Reflection/Runtime/BindingFlagSupport/QueryResult.Enumerator.cs

```diff
@@ -0,0 +1,49 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.CompilerServices;
+
+namespace System.Reflection.Runtime.BindingFlagSupport
+{
+    internal partial struct QueryResult<M> where M : MemberInfo
+    {
+        internal struct QueryResultEnumerator
+        {
+            public QueryResultEnumerator(QueryResult<M> queryResult)
+            {
+                _bindingAttr = queryResult._bindingAttr;
+                _unfilteredCount = queryResult.UnfilteredCount;
+                _queriedMembers = queryResult._queriedMembers;
+                _index = -1;
+            }
+
+            public bool MoveNext()
+            {
+                while (++_index < _unfilteredCount && !_queriedMembers.Matches(_index, _bindingAttr))
+                {
+                }
+
+                if (_index < _unfilteredCount)
+                    return true;
+
+                _index = _unfilteredCount; // guard against wiseguys calling MoveNext() over and over after the end.
+                return false;
+            }
+
+            public M Current
```

Comment:
Is AggressiveInlining necessary here?

```C#
public M Current => _queriedMembers[_index];
```

Reply:
Again, a corert idiom. I'll note these for any possible future sync\sharing.